modules/nw-operator-vsphere-multiple-nic.adoc

```diff
@@ -0,0 +1,75 @@
+
+//Bare metal only?
+[IMPORTANT]
+====
+Each NIC must have access to link:https://quay.io/repository/[Quay.io] to obtain packages that are required for the installation of your cluster.
+====
+
+If you do not configure internal and external CIDR implementations in the {vmw-short} CCM configuration, the {vmw-short} CCM selects the first NIC that the host operating system reports as the primary network interface. This configuration can cause new nodes that associate with a `MachineSet` object with a single NIC to become unusable as each new node receives the `node.cloudprovider.kubernetes.io/uninitialized` taint. In certain situations, the reported network interface might not be the expected network interface. These situations can cause communication issues with the Kubernetes API server that can cause installation of the cluster to fail.
+
+[IMPORTANT]
+====
+If you configured multiple NICs for a network implementation but intend to use only a single NIC for your current needs, ensure that you specify the same IP address for the internal and external CIDR implementations. This configuration ensures that the CCM selects your preferred NIC and not the first NIC that the host operating system reports.
+====
+
+.Example error message because of the {vmw-short} CCM selecting an incompatible NIC
+[source, text]
+----
+ERROR Bootstrap failed to complete: timed out waiting for the condition
+ERROR Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane.
+----
+
+.Prerequisites
+
+* You created Kubernetes manifest files for your {product-title} cluster.
+
+.Procedure
+
+. From the directory where you store your {product-title} cluster manifest files, open the `manifests/cluster-infrastructure-02-config.yml` manifest file.
+
+. Add a `nodeNetworking` object to the file and specify internal and external network subnet CIDR implementations for the object.
++
+.Example of a configured `cluster-infrastructure-02-config.yml` manifest file
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Infrastructure
+metadata:
+  name: cluster
+spec:
+  cloudConfig:
+    key: config
+    name: cloud-provider-config
+  platformSpec:
+    type: VSphere
+    vsphere:
+      failureDomains:
+      - name: generated-failure-domain
+      ...
+      nodeNetworking:
+        external: <1>
+          networkSubnetCidr: 192.168.100.0/24
+        internal: <2>
+          networkSubnetCidr: 172.16.0.0/24
+      vcenters:
+      - datacenters:
+        - sample-dc
+        port: 443
+        server: vcenter.example.com
+status:
+  ...
+  platformStatus:
+    type: VSphere
+----
+<1> The IP address from the CIDR implementation of an external virtual machine network interface.
+<2> The IP address from the CIDR implementation of an internal virtual machine network interface.
```

Comment:
These should be changed as well.  These CIDRs map to the Node `INTERNAL-IP` and `EXTERNAL-IP` fields in `oc get nodes -o wide`.  They should normally be the same.

They aren't assumed to be different interfaces.

https://kubernetes.io/docs/reference/node/node-status/#addresses

- ExternalIP: Typically the IP address of the node that is externally routable (available from outside the cluster).
- InternalIP: Typically the IP address of the node that is routable only within the cluster.

For OCP they are the same.  

We have some weird extra detail here
https://docs.openshift.com/container-platform/4.15/rest_api/node_apis/node-v1.html#status-addresses

But we don't really clarify.

Reply:
Should I mention this command and show example output?

![Screenshot from 2024-06-25 14-10-16](https://github.com/openshift/openshift-docs/assets/57954076/063fe10e-f536-4e1d-b27e-0ec74f3c2e67)

Reply:
`oc get nodes -o wide` is the way to verify.   I patched the `infrastructures` CRD on a live cluster and saw the `EXTERNAL-IP` field update in realtime.

Reply:
Thanks. Added:

> . Enter the following command in the {oc-first} tool to view detailed information, especially `Internal-IP` or `External-IP information, for each node that exists in your cluster. You will need this information in a the `nodeNetworking:` section of the `nodeNetworking` object.
+
[source,terminal]
----
$ oc get nodes -o wide
----